trawler/src/main/java/io/m9sweeper/trawler/framework/scans/Scanner.java

```diff
@@ -74,4 +90,93 @@ default void run() throws RuntimeException {
             throw new RuntimeException(ex);
         }
     }
-}
\ No newline at end of file
+
+    default String getRegistryAuthorization(DockerRegistry registry) throws Exception {
+        String authType = registry.getAuthType();
+        if ("ACR".equals(authType)) {
+            return getACRRegistryAuthorization(registry);
+        } else if ("GCR".equals(authType)) {
+            return getGCRRegistryAuthorization(registry);
+        } else if ("AZCR".equals(authType)) {
+            return getAZCRRegistryAuthorization(registry);
+        } else if (registry.getIsLoginRequired()) {
+            String authPartOne = "export TRIVY_USERNAME=" + escapeXsi(registry.getUsername()) + "; ";
+            String authPartTwo = "export TRIVY_PASSWORD=" + escapeXsi(registry.getPassword()) + "; ";
+            return authPartOne + authPartTwo;
+        }
+        return "";
+    }
+
+    private String escapeXsi(String authToken) {
```

Comment:
All auth related stuff looks generally correct. An unrelated suggestion is that you could make this function protected, so it can be re-used in the classes extending Scanner, since I believe it was used for other things than the auth.

Reply:
Good thought! I'll do that.

Reply:
Update - you can't do that in interfaces. I just made a default method.